File: 2598-smallest-missing-non-negative-integer-after-operations/2598-smallest-missing-non-negative-integer-after-operations.cpp
class Solution {
public:
    int findSmallestInteger(vector<int>& v, int val) {
        unordered_map<int, int> B;
        for (int x : v)
            B[((x % val) + val) % val]++;

        for (int i = 0; i <= v.size(); i++) {
            int rem = i % val;
            if (B.count(rem) && B[rem] > 0)
                B[rem] -= 1;
            else
                return i;
        }
        return 0;
    }
};

File: 2598-smallest-missing-non-negative-integer-after-operations/2598-smallest-missing-non-negative-integer-after-operations.py
class Solution(object):
    def findSmallestInteger(self, nums, value):
        rem = [0] * value
        for x in nums:
            rem[x % value] += 1
        res = 0
        while rem[res % value]:
            rem[res % value] -= 1
            res += 1
        return res